lib/mix/tasks/ecto.load.ex

```diff
@@ -51,29 +52,56 @@ defmodule Mix.Tasks.Ecto.Load do
       (typically in production)
     * `--no-compile` - does not compile applications before loading
     * `--no-deps-check` - does not check depedendencies before loading
-
+    * `--skip-if-loaded` - does not load the dump file if the repo has the migrations table up
   """
 
   @impl true
-  def run(args) do
-    {opts, _} = OptionParser.parse! args, strict: @switches, aliases: @aliases
+  def run(args, table_exists? \\ &Ecto.Adapters.SQL.table_exists?/2) do
+    {opts, _} = OptionParser.parse!(args, strict: @switches, aliases: @aliases)
     opts = Keyword.merge(@default_opts, opts)
 
-    Enum.each parse_repo(args), fn repo ->
+    Enum.each(parse_repo(args), fn repo ->
       ensure_repo(repo, args)
-      ensure_implements(repo.__adapter__, Ecto.Adapter.Structure,
-                                          "load structure for #{inspect repo}")
+      ensure_implements(
+        repo.__adapter__,
+        Ecto.Adapter.Structure,
+        "load structure for #{inspect(repo)}"
+      )
+
+      {:ok, pid, _} = ensure_started(repo, all: true)
```

Comment:
🚨Now the `ensure_started` doesn't exist anymore. How can I ensure that 
repo is up before query `table_exists?` 

I could see that `ensure_started ` was moved to the `Migrator.with_repo/3`, but I don't think calling `Migrator` here would be cool.

Reply:
It is totally fine to call the migrator here. :)

Reply:
Especially because the function we are calling to check for tables is a migration table. :)

Reply:
Done! ✨